### PR TITLE
Add basic typing to the methods in logger

### PR DESCRIPTION
**Pull request by KalleDK**
Added basic typing info, such that people who develops with typing doesn't get error in those lines.

### CODE REVIEW

**Comment by whitej6** [comment]
@KalleDK can you run `invoke tests` locally and resolve and failing tests. The current failing test is from additional whitespace in `bool = False`

Also can you reach out to me via jeremy.white@networktocode.com This PR was considered part of hacktoberfest and would like to show our appreciation for your first time contribution.

**Comment by KalleDK** [comment]
> @KalleDK can you run `invoke tests` locally and resolve and failing tests. The current failing test is from additional whitespace in `bool = False`
> 
> Also can you reach out to me via [jeremy.white@networktocode.com](mailto:jeremy.white@networktocode.com) This PR was considered part of hacktoberfest and would like to show our appreciation for your first time contribution.

Done!

File: nornir_nautobot/utils/logger.py [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
 import logging
 
+from typing import Optional, Any
+
 
 class NornirLogger:
     """Similar to a mixin, to utilize Python logging and Jobs Result obj."""
 
-    def __init__(self, name, nautobot_job=None, debug=False, job_result=None):
+    def __init__(self, name: Optional[str], nautobot_job=None, debug: bool = False, job_result=None):
```
Comment (jvanderaa) [inline nornir_nautobot/utils/logger.py:11]:
```suggestion
    def __init__(self, name: str, nautobot_job=None, debug: bool = False, job_result=None):
```

**Review by jvanderaa — CHANGES_REQUESTED** [review]
This should remain required, not optional.

File: nornir_nautobot/utils/logger.py (hunk 1 above)
Comment (KalleDK) [reply nornir_nautobot/utils/logger.py:11]: Fixing asap

**Review by jvanderaa — APPROVED** [review]

**Pull request merged**